### PR TITLE
[PF-2539] Add update endpoint for flexible resources.

### PR DESCRIPTION
**Pull request by aaron-wang**
### Changes:
- Add an update endpoint
- Allow editing the `data` field of flexible resources
- Assume that `type` and `typeNamespace` are immutable for now

### CODE REVIEW

**Review by ddietterich — APPROVED** [review]
A question about consistent handling of null data inputs.
Not necessary to fix in this PR, but something to think about.

File: service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/any/flight/update/UpdateControlledFlexibleResourceAttributesStep.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+    String newAttributes =
+        DbSerDes.toJson(
+            new FlexibleResourceAttributes(
+                flexResource.getTypeNamespace(), flexResource.getType(), updateData));
+    boolean updated =
+        resourceDao.updateResource(
+            flexResource.getWorkspaceId(),
+            flexResource.getResourceId(),
+            null,
```
Comment (yuhuyoyo) [inline service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/any/flight/update/UpdateControlledFlexibleResourceAttributesStep.java:46]: nit: add comment for the null field so it's clear what they are. e.g. /*param=*/null

File: service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/any/flight/update/UpdateControlledFlexibleResourceAttributesStep.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    }
+    return StepResult.getStepResultSuccess();
+  }
+
+  @Override
+  public StepResult undoStep(FlightContext flightContext) throws InterruptedException {
+    String previousAttributes =
+        flightContext
+            .getWorkingMap()
+            .get(WorkspaceFlightMapKeys.ResourceKeys.PREVIOUS_ATTRIBUTES, String.class);
```
Comment (yuhuyoyo) [inline service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/any/flight/update/UpdateControlledFlexibleResourceAttributesStep.java:62]: if the key doesn't exist, return early?

File: service/src/test/java/bio/terra/workspace/common/utils/MockMvcUtils.java [hunk 3]
```diff
@@ -2398,6 +2466,22 @@ public void postExpect(
         .andExpect(status().is(httpStatus));
   }
 
+  /** Patch http request and expect error thrown. */
+  public ResultActions patchExpect(
```
Comment (yuhuyoyo) [inline service/src/test/java/bio/terra/workspace/common/utils/MockMvcUtils.java:2470]: there is an updateresource method already in MockMvcUtils. can you use that?

**Review by yuhuyoyo — APPROVED** [review]

**Pull request merged**